src/ol_infrastructure/substructure/keycloak/__main__.py

```diff
@@ -224,24 +207,44 @@
     "ol-apps-update-password",
     realm_id=ol_apps_realm.realm,
     alias="UPDATE_PASSWORD",
-    default_action=True,
+    default_action=False,
     enabled=True,
     opts=resource_options,
 )
 
-ol_apps_touchstone_saml_identity_provider = keycloak.saml.IdentityProvider(
-    "touchstone-idp",
-    realm=ol_apps_realm.id,
-    alias="touchstone-idp",
-    display_name="MIT Touchstone",
-    entity_id=f"{keycloak_url}/realms/olapps",
-    force_authn=True,
-    post_binding_response=True,
-    principal_attribute="urn:oid:1.3.6.1.4.1.5923.1.1.1.6",
-    single_sign_on_service_url="https://idp.mit.edu/idp/profile/SAML2/Redirect/SSO",
-    trust_email=True,
-    validate_signature=True,
-    want_assertions_encrypted=True,
-    want_assertions_signed=True,
-    opts=resource_options,
-)
+
+# Check if any Openid clients exist in config and create them
+for openid_clients in keycloak_config.get_object("openid_clients"):
+    realm_name = openid_clients.get("realm_name")
+    client_details = openid_clients.get("client_info")
+    for client_name, client_url in client_details.items():
+        openid_client = keycloak.openid.Client(
+            f"ol-{client_name}-client",
+            realm_id=realm_name,
+            client_id=f"ol-{client_name}-client",
```

Comment:
Maybe include the realm in the client_id as noted above.

Reply:
Not sure if it's useful to add realm_name here as the clients in keycloak will be listed under the individual realms, so it'll be clear what realm they belong to. Not sure it gives us anything to add it here.

Reply:
Makes sense.